src/util/serialize.h

```diff
@@ -426,13 +498,19 @@ MAKE_STREAM_WRITE_FXN(s32,   S32,      4);
 MAKE_STREAM_WRITE_FXN(s64,   S64,      8);
 MAKE_STREAM_WRITE_FXN(f32,   F1000,    4);
 MAKE_STREAM_WRITE_FXN(f32,   F32,      4);
+MAKE_STREAM_WRITE_FXN(double,   F32,      4);
+MAKE_STREAM_WRITE_FXN(double,   D1000,    4);
+MAKE_STREAM_WRITE_FXN(double,   D,      8);
```

Comment:
Also regarding naming: To follow the convention, `writeD` could be named `writeF64` (and `writeV3D` `writeV3F64`)

Reply:
Great point. I replaced D with F64 throughout.